dev/integration_tests/flutter_gallery/test_driver/transitions_perf_e2e.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:e2e/e2e.dart';
+
+import 'package:flutter_gallery/gallery/app.dart' show GalleryApp;
+import 'package:flutter_gallery/gallery/demos.dart';
+import 'package:flutter_gallery/demo_lists.dart';
+
+import 'e2e_util.dart' show watchPerformance;
+
+const List<String> kSkippedDemos = <String>[];
+
+// All of the gallery demos, identified as "title@category".
+//
+// These names are reported by the test app, see _handleMessages()
+// in transitions_perf.dart.
+List<String> _allDemos = kAllGalleryDemos.map(
+  (GalleryDemo demo) => '${demo.title}@${demo.category.name}',
+).toList();
+
+/// Scrolls each demo menu item into view, launches it, then returns to the
+/// home screen twice.
+Future<void> runDemos(List<String> demos, WidgetTester tester) async {
```

Comment:
Two things: `pumpAndSettle` and `pageBack`: 
- `pumpAndSettle` can be replaced by `pump` for a duration, and with policy not being `fullyLive` (which is fine because we don't need accurate control of time but want to wait the frame). The issue is, wait for settling in flutter driver I believe is part of adb (or its equivalence of iOS debugger), while here we are bound to estimate the time, resulting in potentially extra frames, meaning the metric may not be a faithful equivalence of the original driver test; 
- `pageBack` can be implemented separately or promote to `WidgetController`. To promote, I need to delete this line: https://github.com/flutter/flutter/blob/30aef0a3b9611763f8e60985e7cca9cb30c1ea6a/packages/flutter_test/lib/src/widget_tester.dart#L1072 which I'm not sure if I should.

Reply:
Or alternatively, make `pumpAndSettle` an abstract method in `WidgetController` and implement it in `LiveWidgetController`. The tricky part is that there's a `EnginePhase phase` optional order/un-named argument in `pumpAndSettle` in the middle of `duration` and `timeout`. `duration` and `timeout` should be part of the API but `phase` only makes sense for `WidgetTester`. This will be a breaking change but luckily ALL `pumpAndSettle` calls in the repo doesn't use this argument.

Reply:
A demo of the change #62640

Reply:
Done with the newly merged `pumpAndSettle`.